Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+        private var horizontalAccuracy: Double?
+        
+        /// An indoors location data source based on sensor data, including but not
+        /// limited to radio, GPS, motion sensors.
+        private var indoorsLocationDataSource: IndoorsLocationDataSource?
+        
+        /// The map's location display.
+        let locationDisplay = LocationDisplay(dataSource: SystemLocationDataSource())
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextLeading: String = ""
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextTrailing: String = ""
+        
+        /// Represents loading state of indoors data, blocks interaction until loaded.
+        @Published private(set) var isLoading = false
+        
+        /// Kicks off the logic loading the data for the indoors map and indoors location.
+        func loadIndoorData() async throws {
+            isLoading = true
+            defer { isLoading = false }
+            try await map.load()
+            try await setIndoorDatasource()
+        }
+        
+        /// Stops the location data source.
+        func stopLocationDataSource() {
+            Task {
+                await locationDisplay.dataSource.stop()
+            }
+        }
+        
+        /// Sets the indoor datasource on the location display depending on
+        /// whether the map contains an IndoorDefinition.
+        private func setIndoorDatasource() async throws {
+            try await map.floorManager?.load()
+            // If an indoor definition exists in the map, it gets loaded and sets the IndoorsDataSource to pull information
+            // from the definition, otherwise the IndoorsDataSource attempts to create itself using IPS table information.
+            if let indoorPositioningDefinition = map.indoorPositioningDefinition {
+                try await indoorPositioningDefinition.load()
+                indoorsLocationDataSource = IndoorsLocationDataSource(definition: indoorPositioningDefinition)
+            } else {
+                indoorsLocationDataSource = try await createIndoorLocationDataSourceFromTables(map: map)
+            }
+            // This ensures that the details of the inside of the building, like room layouts are displayed.
+            for featLayer in map.operationalLayers {
+                if featLayer.name == "Transitions" || featLayer.name == "Details" {
+                    featLayer.isVisible = true
+                }
+            }
+            // The indoorsLocationDataSource should always be there. Since the createIndoorLocationDataSourceFromTables returns
+            // an optional value it cannot be guaranteed. Best option if you get to this point without a datasource
+            // is to return (ideally an error would have been thrown before this point and the flow broken.)
```

Comment:
Since one of the goals of samples is to model best practices for our customers, we should do what is ideal if at all possible.

Reply:
So should I steer this sample code further away from what is in the indoors developer documentation for iOS?